crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1944,9 +1944,18 @@ impl<'src> Parser<'src> {
     /// See: <https://docs.python.org/3/reference/expressions.html#list-displays>
     fn parse_list_like_expression(&mut self) -> Expr {
         let start = self.node_start();
-
         self.bump(TokenKind::Lsqb);
 
+        if self.at(TokenKind::Unknown)
+            && (self.peek() == TokenKind::EndOfFile || self.peek() == TokenKind::Newline)
```

Comment:
While this works for `[` at the end of a file, it doesn't help if we have something like this


```py
a = [

if True:
	pass
```

I think we want to use something like `if self.at_sequence_end() {` or similar instead.